mne/time_frequency/tfr.py

```diff
@@ -1107,9 +1162,240 @@ def plot(self, picks, baseline=None, mode='mean', tmin=None, tmax=None,
+            from scipy.signal import argrelmax
+            order = tf_data.shape[2] // 30
+            if order < 1:
+                order = 1
+            peaks_idx = argrelmax(tf_data, order=order, axis=2)
+            if peaks_idx[0].size == 0:
+                _, p_t, p_f = np.unravel_index(tf_data.argmax(),
+                                               tf_data.shape)
+                timefreqs = [(tf_times[p_t], tf_freqs[p_f])]
+            else:
+                peaks = [tf_data[0, f, t] for f, t in
+                         zip(peaks_idx[1], peaks_idx[2])]
+                peakmax_idx = np.argmax(peaks)
+                peakmax_time = tf_times[peaks_idx[2][peakmax_idx]]
+                peakmax_freq = tf_freqs[peaks_idx[1][peakmax_idx]]
+
+                timefreqs = [(peakmax_time, peakmax_freq)]
+
+        vmin, vmax = _setup_vmin_vmax(tf_data, vmin, vmax)
+
+        # topomap
+        contours = topomap_args.get('contours', 6)
+        locator, contours = _set_contour_locator(vmin, vmax, contours)
+
+        topomap_args_pass = {k: v for k, v in topomap_args.items() if
+                             k not in ('axes', 'show', 'colorbar')}
+        topomap_args_pass['outlines'] = (topomap_args['outlines'] if 'outlines'
+                                         in topomap_args else 'skirt')
+        topomap_args_pass['contours'] = contours
+
+        for idx in range(len(map_ax)):
+
+            time = timefreqs[idx][0]
+            freq = timefreqs[idx][1]
+
+            if isinstance(timefreqs_average, tuple):
+                time_half_range = timefreqs_average[0]/2
+                freq_half_range = timefreqs_average[1]/2
+            elif isinstance(timefreqs_average, list):
+                time_half_range = timefreqs_average[idx][0]/2
+                freq_half_range = timefreqs_average[idx][1]/2
+            else:
+                time = self.times[np.argmin(np.abs(self.times - time))]
+                freq = self.freqs[np.argmin(np.abs(self.freqs - freq))]
+                time_half_range = freq_half_range = 0
+
+            tmin, tmax = time - time_half_range, time + time_half_range
+            fmin, fmax = freq - freq_half_range, freq + freq_half_range
+
+            if (time_half_range == 0) and (freq_half_range == 0):
+                map_ax[idx].set_title('(%d ms,\n%.1f Hz)' % (time * 1e3, freq))
+            else:
+                map_ax[idx].set_title('(%d \u00B1 %d ms,\n%.1f \u00B1 %.1f '
+                                      'Hz)' % (time * 1e3,
+                                               time_half_range * 1e3,
+                                               freq, freq_half_range))
+
+            self.plot_topomap(tmin=tmin, tmax=tmax, fmin=fmin, fmax=fmax,
+                              axes=map_ax[idx], baseline=baseline, mode=mode,
+                              layout=layout, vmin=vmin, vmax=vmax, cmap=cmap,
+                              colorbar=False, show=True, **topomap_args_pass)
```

Comment:
I think you should use ``show=False``, to prevent showing the figure prematurely and call ``plt_show`` at the end (like you're doing).